### PR TITLE
Ben/add mlm objective

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/base.py [hunk 1]
```diff
@@ -386,6 +387,9 @@ def generate_text(self, seed_text='', max_length=None, use_extra_toks=True):
         :param seed_text: Defaults to the empty string. This will form the starting point to begin modelling
         :return: A string containing the generated text.
         """
+        if self.config.mlm:
+            raise FinetuneError("Cannot generate language with a Mass Language Model")
```
Comment (madisonmay) [inline finetune/base.py]: Mass language model :thinking:

File: finetune/datasets/summaries_unsupervised_mlm.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import logging
+
+from newsroom import jsonl
+import random
+from finetune import Classifier
+import ujson as _json
+import glob
+
+logging.basicConfig(level=logging.DEBUG)
+DATASET_PATH = "/root/code/Data/newsroom/train_half.data"
```
Comment (madisonmay) [inline finetune/datasets/summaries_unsupervised_mlm.py]: If we do ship this in the datasets folder I think we should probably train on something small / naive, maybe with trimmed down model size as a toy example.

Comment (madisonmay) [reply finetune/datasets/summaries_unsupervised_mlm.py]: I would feel weird redistributing a portion of the summaries dataset... need to at least check licensing.

File: finetune/transformer.py [hunk 3]
```diff
@@ -38,8 +38,12 @@ def _attn(q, k, v, attn_pdrop, train=False, scale=False, mask=True):
         n_state = shape_list(v)[-1]
         w = w * tf.rsqrt(tf.cast(n_state, tf.float32))
 
-    if mask:
+    if mask is True:
         w = mask_attn_weights(w)
+
+    elif isinstance(mask, tf.Tensor):
```
Comment (madisonmay) [inline finetune/transformer.py]: Wait, what's going on here?

**Comment by madisonmay** [comment]
Add shit check to ensure training w/ MLM objective doesn't error out.

**Comment by khoa-ho** [comment]
Do you guys have a timeline for this? Being able to easily finetune BERT LM will be game-changing!

**Pull request closed without merging**